src/main/java/com/puppycrawl/tools/checkstyle/checks/header/AbstractHeaderCheck.java

```diff
@@ -60,8 +60,7 @@ public abstract class AbstractHeaderCheck extends AbstractFileSetCheck
 
     /** Specify the character encoding to use when reading the headerFile. */
     @XdocsPropertyType(PropertyType.STRING)
-    private Charset charset = createCharset(System.getProperty("file.encoding",
-        StandardCharsets.UTF_8.name()));
+    private final Charset charset = createCharset(StandardCharsets.UTF_8.name());
```

Comment:
this system property might be tricky
https://stackoverflow.com/a/362006

Reply:
what it mean is that we become to ignore this system property, and just use UTF-8 by default.
I am ok, as we have setCharset property to be explicit from our config.

But I think we need to create special issue on this to let users see it release notes as explicit deprecation, not a some pitest refactoring.

Reply:
discussion happens at https://github.com/checkstyle/checkstyle/pull/13268#discussion_r1249956750 for this problem